### PR TITLE
Add script to populate Algolia for search

### PR DESCRIPTION
**Pull request by jacobvenable**
This PR does not close an issue.

This PR should be merged after #73 since this will change the name of the properties pulled off of a resource as well as the name of the file containing resources.

## What does this PR do?

This creates a new npm script that runs a new `algolia-update` utility. This utility will then:

1. read `data_pipeline/charity.json`
2. pull create an array of objects with each object containing an individual resource's:
    - `charityname`
    - `category`
    - `servicetype`
    - `charityID` (renamed `objectID` as [required by Algolia](https://www.algolia.com/doc/api-client/methods/indexing/?language=php#the-object-id))
3. send that array of objects to Algolia to populate the given index

## Next

Using the populated index in Algolia, we'll implement a `withSearchResults` HOC to pull data from Algolia.

### CODE REVIEW

File: data_pipeline/algolia-update.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+//load local environment variables
+const env = {
+  local: dotenv.parse(fs.readFileSync('.env.local'))
+};
+
+const charities = JSON.parse(
+  fs.readFileSync('data_pipeline/charity.json').toString('utf8')
+);
+
+const updatedAlgoliaIndex = [];
```
Comment (rhinodavid) [inline data_pipeline/algolia-update.js]:
```
const updatedAlgoliaIndex = Object.entries(charities).map(
  (result, [id, {category, charityname, servicetype]) => ({
    category,
    charityname,
    objectId: id,
    servicetype,
  )}
);
```
for the big kids

File: data_pipeline/algolia-update.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const fs = require('fs');
+const dotenv = require('dotenv');
+const algoliasearch = require('algoliasearch');
```
Comment (rhinodavid) [inline data_pipeline/algolia-update.js]:
```suggestion
const algoliaSearch = require('algoliasearch');
```

File: data_pipeline/algolia-update.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      objectID: charityID, //objectID required by Algolia
+      charityname,
+      category,
+      servicetype
+    });
+  }
+}
+
+// initialize Algolia project
+const algolia = algoliasearch(
```
Comment (rhinodavid) [inline data_pipeline/algolia-update.js]:
```suggestion
const algolia = algoliaSearch(
```

File: package.json [hunk 4]
```diff
@@ -5,26 +5,28 @@
...
+    "react-router": "^4.3.1",
     "react-router-config": "^4.4.0-beta.6",
     "react-router-dom": "^4.3.1",
-    "react-router": "^4.3.1",
     "react-scripts": "2.1.5",
-    "react": "^16.8.3",
     "typescript": "^3.3.4000"
   },
   "scripts": {
+    "algolia": "node data_pipeline/algolia-update.js",
```
Comment (rhinodavid) [inline package.json]: `algolia-update`? `update-algolia`

File: data_pipeline/algolia-update.js (hunk 2 above)
Comment (jacobvenable) [reply data_pipeline/algolia-update.js]: 👍 [updated](https://github.com/CodeForBoulder/upswyng/pull/74/commits/aa76a218639abf16ba1dfcd8cb713756035ddc1c)

File: data_pipeline/algolia-update.js (hunk 3 above)
Comment (jacobvenable) [reply data_pipeline/algolia-update.js]: 👍 [updated](https://github.com/CodeForBoulder/upswyng/pull/74/commits/aa76a218639abf16ba1dfcd8cb713756035ddc1c)

File: package.json (hunk 4 above)
Comment (jacobvenable) [reply package.json]: good point. [updated](https://github.com/CodeForBoulder/upswyng/pull/74/commits/aa97f5b5a5abaadd52d0984d584373d2cf51e1ef)

File: data_pipeline/algolia-update.js (hunk 1 above)
Comment (jacobvenable) [reply data_pipeline/algolia-update.js]: mmm good call. [updated](https://github.com/CodeForBoulder/upswyng/pull/74/commits/f3552742c7ceb1074f2d9bbafb397f6bb8142c46)

File: data_pipeline/algolia-update.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+//load local environment variables
+const env = {
+  local: dotenv.parse(fs.readFileSync('.env.local'))
+};
+
+const charities = JSON.parse(
+  fs.readFileSync('data_pipeline/charity.json').toString('utf8')
+);
+
+const updatedAlgoliaIndex = Object.entries(charities).map(
```
Comment (rhinodavid) [inline data_pipeline/algolia-update.js]: how does this make you feel leave giffy below

File: data_pipeline/algolia-update.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  ([charityID, { charityname, category, servicetype }]) => {
+    return {
```
Comment (rhinodavid) [inline data_pipeline/algolia-update.js]: can shorthand this

Comment (jacobvenable) [reply data_pipeline/algolia-update.js]: [upderped](https://github.com/CodeForBoulder/upswyng/pull/74/commits/0fe944f5aed3d51015382d6fdea052a981c9a248)

**Comment by jacobvenable** [comment]
> Don't we want to throw the resource description in the index too?

We'll worry about what actually get placed in the search index a bit later, but I could definitely see being added eventually. Will want to work with Mike about adding the items that should be there.

**Pull request merged**